api/app/Policies/PoolPolicy.php

```diff
@@ -131,16 +125,14 @@ public function updateDraft(User $user, Pool $pool)
     public function publish(User $user, Pool $pool)
     {
         // The status must be DRAFT to be able to publish it.
-        if ($pool->getStatusAttribute() !== PoolStatus::DRAFT->name) {
-            return Response::deny('Pool has already been published.');
+        if ($pool->getStatusAttribute() === PoolStatus::DRAFT->name) {
+            // The closing date must be greater than today's date at the end of day.
+            if ($pool->closing_date && $pool->closing_date > Carbon::now()->endOfDay()) {
+                return $user->isAbleTo('publish-any-pool');
+            }
         }
 
-        // The closing date must be greater than today's date at the end of day.
-        if ($pool->closing_date && $pool->closing_date < Carbon::now()->endOfDay()) {
-            return Response::deny('Expiry date must be a future date.');
-        }
-
-        return $user->isAbleTo('publish-any-pool');
+        return false;
```

Comment:
As above, its okay to keep using `else` statements to return a useful `Response::deny`

Reply:
I've (re)added some error responses with [Update pool publish policy with specific denial messages](https://github.com/GCTC-NTGC/gc-digital-talent/pull/8312/commits/4d517a84e304ac1373deecf0cfc01031f30eaad7).